test.js

```diff
@@ -45,3 +45,30 @@ test('can be aborted', async t => {
 	t.true(error instanceof pThrottle.AbortError);
 	t.true(end() < 100);
 });
+
+test('`this` is preserved in throttled function', async t => {
+	class FixtureClass {
+		constructor() {
+			this._foo = fixture;
+		}
+
+		foo() {
+			// If `this` is not preserved by `pThrottle()`
+			// then `this` will be undefined and accesing `this._foo` will throw.
+			return this._foo;
+		}
+
+		getThis() {
+			// If `this` is not preserved by `pThrottle()`
+			// then `this` will be undefined.
+			return this;
+		}
+	}
+	FixtureClass.prototype.foo = pThrottle({limit: 1, interval: 100})(FixtureClass.prototype.foo);
+	FixtureClass.prototype.getThis = pThrottle({limit: 1, interval: 100})(FixtureClass.prototype.getThis);
+
+	const thisFixture = new FixtureClass();
+
+	t.is(await thisFixture.getThis(), thisFixture);
+	await t.notThrowsAsync(() => thisFixture.foo());
```

Comment:
This should not use async assertion.

Reply:
Are you sure? 

`73:2  Use await with t.notThrowsAsync().  ava/use-t-throws-async-well`

```typescript
/** Assert that the async function does not throw. You must await the result. */
(fn: () => PromiseLike<any>, message?: string): Promise<void>;
```

Reply:
after `FixtureClass.prototype.foo = pThrottle({limit: 1, interval: 100})(FixtureClass.prototype.foo);`, `thisFixture.foo()` returns a Promise.  So I am waiting for it.

Reply:
Oh, I missed that. But you're using the assertion incorrectly. It accepts a `Promise`, so you can drop the function.

Reply:
Are you saying:
```javascript
await t.notThrowsAsync(thisFixture.foo);
```

If so, I don't think that's quite right.  Doing so means the `this` context is lost.   By wrapping it in fn, it is clear that `foo()` is bound to `thisFixture`

Reply:
```js
await t.notThrowsAsync(thisFixture.foo());
```

Reply:
Thanks - I missed the second form when reading its interface.  I made this change.

```typescript
(fn: () => PromiseLike<any>, message?: string): Promise<void>;
/** Assert that the promise does not reject. You must await the result. */
(promise: PromiseLike<any>, message?: string): Promise<void>;
```